Framework/Subset_of_CorLib/System/Single.cs

```diff
@@ -22,14 +22,31 @@ public struct Single
 
         public override String ToString()
         {
-            return Number.Format(m_value, false, "G", NumberFormatInfo.CurrentInfo);
+            string str = ((Double)m_value).ToString();

```

Comment:
This seems kinda wonky.... Shouldn't the fic be done in Number.Format directly instead of here? Isn't it possible something else calls Number.Format() that would still get the incorrect behavior?